src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Analyzer.cs

```diff
@@ -0,0 +1,205 @@
+                return;
+            }
+
+            IMethodSymbol? toUpperParameterlessMethod = stringType.GetMembers(StringToUpperMethodName).OfType<IMethodSymbol>().GetFirstOrDefaultMemberWithParameterInfos();
+            if (toUpperParameterlessMethod == null)
+            {
+                return;
+            }
+
+            IMethodSymbol? toUpperInvariantParameterlessMethod = stringType.GetMembers(StringToUpperInvariantMethodName).OfType<IMethodSymbol>().GetFirstOrDefaultMemberWithParameterInfos();
+            if (toUpperInvariantParameterlessMethod == null)
+            {
+                return;
+            }
+
+            // Retrieve the diagnosable string overload methods: Contains, IndexOf, StartsWith, CompareTo
+
+            ParameterInfo[] stringParameter = new[]
+            {
+                ParameterInfo.GetParameterInfo(stringType)
+            };
+
+            IMethodSymbol? containsStringMethod = stringType.GetMembers(StringContainsMethodName).OfType<IMethodSymbol>().GetFirstOrDefaultMemberWithParameterInfos(stringParameter);
+            if (containsStringMethod == null)
+            {
+                return;
+            }
+
+            // TODO: There are more overloads that take StringComparison, diagnose only the simple one for now
+            IMethodSymbol? indexOfStringMethod = stringType.GetMembers(StringIndexOfMethodName).OfType<IMethodSymbol>().GetFirstOrDefaultMemberWithParameterInfos(stringParameter);
+            if (indexOfStringMethod == null)
+            {
+                return;
+            }
+
+            IMethodSymbol? startsWithStringMethod = stringType.GetMembers(StringStartsWithMethodName).OfType<IMethodSymbol>().GetFirstOrDefaultMemberWithParameterInfos(stringParameter);
+            if (startsWithStringMethod == null)
+            {
+                return;
+            }
+
+            IMethodSymbol? compareToStringMethod = stringType.GetMembers(StringCompareToMethodName).OfType<IMethodSymbol>().GetFirstOrDefaultMemberWithParameterInfos(stringParameter);
+            if (compareToStringMethod == null)
+            {
+                return;
+            }
+
+            // Retrieve the StringComparer properties that need to be flagged: CurrentCultureIgnoreCase, InvariantCultureIgnoreCase
+
+            IEnumerable<IPropertySymbol> ccicPropertyGroup = stringComparerType.GetMembers(StringComparisonCurrentCultureIgnoreCaseName).OfType<IPropertySymbol>();
+            if (!ccicPropertyGroup.Any())
+            {
+                return;
+            }
+
+            IEnumerable<IPropertySymbol> icicPropertyGroup = stringComparerType.GetMembers(StringComparisonInvariantCultureIgnoreCaseName).OfType<IPropertySymbol>();
+            if (!icicPropertyGroup.Any())
+            {
+                return;
+            }
+
+            context.RegisterOperationAction(context =>
+            {
+                IInvocationOperation caseChangingInvocation = (IInvocationOperation)context.Operation;
```

Comment:
```suggestion
                var caseChangingInvocation = (IInvocationOperation)context.Operation;
```

Reply:
I prefer to use explicit.

Reply:
Which made me realize I accidentally used `var` in a couple of places. For consistently, I'll use explicit in those locations.